### PR TITLE
Badge fixes

### PR DESCRIPTION
**Pull request by riknoll**
Fixes https://github.com/microsoft/pxt-arcade/issues/4286
Fixes https://github.com/microsoft/pxt-arcade/issues/4290
Fixes https://github.com/microsoft/pxt-arcade/issues/4295

Also does a ton of css tweaks based on feedback @soniakandah sent me!

### CODE REVIEW

**Review by jwunderl — APPROVED** [review]

**Review by livcheerful — APPROVED** [review]

File: react-common/components/profile/BadgeInfo.tsx [hunk 1]
```diff
@@ -15,24 +15,24 @@ export const BadgeInfo = (props: BadgeInfoProps) => {
...
-        }
-        { badge.timestamp ?
             <div className="profile-badge-info-text">
-                {date.toLocaleDateString(pxt.U.userLanguage())}
+                {badgeDescription(badge)}
             </div>
-            : undefined
-        }
+        </div>
+        { badge.timestamp ?
```
Comment (shakao) [inline react-common/components/profile/BadgeInfo.tsx:26]: small: `badge.timestamp &&` ?

Comment (riknoll) [reply react-common/components/profile/BadgeInfo.tsx:26]: timestamp is a number, so if it's zero it results in a 0 being printed. I think I was initializing it to 0 at some point?

Comment (shakao) [reply react-common/components/profile/BadgeInfo.tsx:26]: could do !== undefined or something but it's also not a huge deal

File: react-common/styles/profile/profile.css [hunk 2]
```diff
@@ -104,6 +115,11 @@
     box-shadow: none;
 }
 
+.profile-badge-subtitle {
+    font-style: italic;
+    color: #aba9a5;
```
Comment (shakao) [inline react-common/styles/profile/profile.css:120]: should this be in a less variable?

File: skillmap/src/lib/skillMapParser.ts [hunk 3]
```diff
@@ -336,6 +336,9 @@ function inflateMapReward(section: MarkdownSection, base: Partial<MapRewardNode>
             }
         }
 
+        const priority = ["completion-badge", "certificate"];
```
Comment (shakao) [inline skillmap/src/lib/skillMapParser.ts:339]: we're not going to let skillmap authors determine their own reward order?

File: skillmap/src/components/AppModal.tsx [hunk 4]
```diff
@@ -419,77 +421,126 @@ export class AppModalImpl extends React.Component<AppModalProps, AppModalState>
...
+                    <span>{lf("Next Reward")}</span>
+                    <i className="icon right circle arrow" />
+                </button>
+            )
         }
 
-        return <Modal title={title} actions={buttons} onClose={this.handleOnClose}>
+        return <Modal title={title} onClose={this.handleOnClose}>
             {lf("Use the button below to get your completion certificate!")}
+            <div className="reward-buttons">
```
Comment (shakao) [inline skillmap/src/components/AppModal.tsx:469]: are we doing some special/different design for the reward modal buttons? can we make these changes across all the skillmap modal buttons? don't love moving them out like this, "open reward link in new tab" and "move to the next modal" both still feel like modal actions to me

Comment (riknoll) [reply skillmap/src/components/AppModal.tsx:469]: We were already doing that for the completion modal, I'm just reusing the css from the button in that modal

File: react-common/styles/profile/profile.css (hunk 2 above)
Comment (riknoll) [reply react-common/styles/profile/profile.css:120]: I'll move all colors into variables

File: skillmap/src/components/AppModal.tsx (hunk 4 above)
Comment (shakao) [reply skillmap/src/components/AppModal.tsx:469]: the completion modal also has separate follow-up actions though, these are the follow-up actions for this modal. this is more a design question than an implementation one, i guess. could we pass in the modal button class as a property to the actions?

**Review by eanders-ms — APPROVED** [review]

**Review by shakao — APPROVED** [review]

**Review by shakao — APPROVED** [review]

**Pull request merged**